### PR TITLE
added Atmospheric analyser

### LINKED ISSUE #2313: Atmospheric Analyzer
# Description
Implement the atmospheric analyzer, matching the way it works from tg.

Basically, it should just output a message showing the current gas contents of the tile you are on.

### PR DESCRIPTION
**Pull request by Bod9001**
# Pull Request Template
and maped them
Fixes #2313 

Tested in Editor,
 uses the same logic as the atmospheric analyser so Should be okay?

### CODE REVIEW

File: UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AtmosphericAnalyser : MonoBehaviour, ICheckedInteractable<HandActivate>
+{
+
+	public bool WillInteract(HandActivate interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side))
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs]: FYI you don't even need to implement WillInteract if you're just going to call the default validation. If you instead implement IInteractable<HandActivate> you can omit WillInteract and it'll call the default validation automatically.

File: UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		}
+
+		return true;
+	}
+
+	public void ServerPerformInteraction(HandActivate interaction)
+	{
+		string ToShow = "";
+		var MetaDataLayer = MatrixManager.AtPoint(interaction.PerformerPlayerScript.registerTile.WorldPositionServer, true).MetaDataLayer;
+		if (MetaDataLayer != null) {
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs]:
```suggestion
		if (MetaDataLayer != null) 
		{ 
```

File: UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		return true;
+	}
+
+	public void ServerPerformInteraction(HandActivate interaction)
+	{
+		string ToShow = "";
+		var MetaDataLayer = MatrixManager.AtPoint(interaction.PerformerPlayerScript.registerTile.WorldPositionServer, true).MetaDataLayer;
+		if (MetaDataLayer != null) { 
+			var Node = MetaDataLayer.Get(interaction.Performer.transform.localPosition.RoundToInt());
+			if(Node != null){
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs]:
```suggestion
			if(Node != null)
			{
```

File: UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+				ToShow = "Pressure : " + Node.GasMix.Pressure + " Kpa \n"
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs]: When I use atmos scanner out in space I get "NaN%" showing up for some of these. I think it should say 0 instead.

**Review by chairbender — CHANGES_REQUESTED** [review]
Nice to have this in! Might even make debugging atmos a bit easier so we can see what's happening on the headless server.

You should see if you can get your IDE auto-format to respect the code style (curly braces on newline) so it can be more consistent across the project. Would also be nice if it can ensure you use lowercase for local variables (ToShow, MetaDataLayer and Node).

File: UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs (hunk 4 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs]: if you want to use it for debugging it should show NaN%?

Comment (Jibbajabbafic) [reply UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs]: Well it's an in-game item which is gonna get used by players too

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs]: should I just replace it with non-existent?

**Comment by Bod9001** [comment]
mapping

File: UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs (hunk 4 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs]: is 0 now

**Pull request merged**